library/dnet.c

```diff
@@ -1013,6 +1484,9 @@ static int dnet_process_cmd_with_backend_raw(struct dnet_backend_io *backend, st
 	gettimeofday(&end, NULL);
 	diff = DIFF(start, end);
 
+	/* If there was any error - send ACK to notify client with error code and destroy transaction */
+	if (err)
+		cmd->flags |= DNET_FLAGS_NEED_ACK;
```

Comment:
By default, all handlers send ACK with error by themselves. READ, WRITE and LOOKUP were exceptions because they could be handled by differently (by cache or by backend) - it was solved via expecting that high level function (`dnet_process_cmd_with_backend_raw`) will send ACK for them.

With these changes, all commands shouldn't send ACK by themselves, so it should be carefully rechecked.